pkg/network/ebpf/c/kafka/kafka-helpers.h

```diff
@@ -0,0 +1,200 @@
+#ifndef __KAFKA_HELPERS_H
+#define __KAFKA_HELPERS_H
+
+#include "kafka-types.h"
+
+// Forward declaration
+static __always_inline bool try_parse_produce_request(kafka_transaction_t *kafka_transaction);
+static __always_inline bool try_parse_fetch_request(kafka_transaction_t *kafka_transaction);
+static __always_inline bool extract_and_set_first_topic_name(kafka_transaction_t *kafka_transaction);
+
+static __inline int32_t read_big_endian_int32(const char* buf) {
+    int32_t val;
+    bpf_probe_read_kernel(&val, 4, (void*)buf);
+    return bpf_ntohl(val);
+}
+
+static __inline int16_t read_big_endian_int16(const char* buf) {
+    int16_t val;
+    // De-Referencing buf causes misalignment verifier error for unknown reason, so using bpf_probe_read_kernel as a workaround
+    bpf_probe_read_kernel(&val, 2, (void*)buf);
+    return bpf_ntohs(val);
+}
+
+static __inline int32_t kafka_read_big_endian_int32(kafka_transaction_t *kafka_transaction) {
+    const int32_t value = read_big_endian_int32(kafka_transaction->request_fragment + kafka_transaction->current_offset_in_request_fragment);
+    kafka_transaction->current_offset_in_request_fragment += 4;
+    return value;
+}
+
+static __inline int16_t kafka_read_big_endian_int16(kafka_transaction_t *kafka_transaction) {
+    const int16_t value = read_big_endian_int16(kafka_transaction->request_fragment + kafka_transaction->current_offset_in_request_fragment);
+    kafka_transaction->current_offset_in_request_fragment += 2;
+    return value;
+}
+
+// Checking if the buffer represents kafka message
+static __always_inline bool try_parse_request_header(kafka_transaction_t *kafka_transaction) {
+    char *request_fragment = kafka_transaction->request_fragment;
+    if (request_fragment == NULL) {
+        return false;
+    }
+
+    const int32_t message_size = kafka_read_big_endian_int32(kafka_transaction);
+    log_debug("message_size: %d", message_size);
+    if (message_size <= 0) {
+        return false;
+    }
+
+    const int16_t request_api_key = kafka_read_big_endian_int16(kafka_transaction);
+    log_debug("request_api_key: %d", request_api_key);
+    if (request_api_key < 0 || request_api_key > KAFKA_MAX_VERSION) {
+        log_debug("request_api_key < 0 || request_api_key > KAFKA_MAX_VERSION");
+        return false;
+    }
```

Comment:
It feels to me that we should definitely be more judicious about what we consider to be of interest here since this will impact how many events we send to userspace.

|Name|Key|
|---|---|
|Produce |	0|
|Fetch |	1|
|Offsets |	2|
|Metadata |	3|
|LeaderAndIsr |	4|
|StopReplica |	5|
|UpdateMetadata |	6|
|ControlledShutdown |	7|
|OffsetCommit |	8|
|OffsetFetch |	9|
|GroupCoordinator |	10|
|JoinGroup |	11|
|Heartbeat |	12|
|LeaveGroup |	13|
|SyncGroup |	14|
|DescribeGroups |	15|
|ListGroups |	16|

Is it fair to assume that _for now_ we're only interested in `Produce` and `Fetch` messages?
Probably worth creating an `enum` for this and extracting this validation to a helper function

Reply:
UPDATE: I just realized that you're pretty much doing all of that later on in the code :man_facepalming: 
In this case my only recommendation is to short-circuit the code earlier for messages we're not interested in
Like, instead of `(request_api_key < 0 || request_api_key > KAFKA_MAX_VERSION)` you can change this to 
`(request_api_key != KAFKA_FETCH && request_api_key != KAFKA_PRODUCE)` or something like that